ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -123,26 +115,13 @@ public void requestInterstitialAd(Context context,
     final String gameId = serverParameters.getString(KEY_GAME_ID);
     mPlacementId = serverParameters.getString(KEY_PLACEMENT_ID);
 
-    if (!isValidIds(gameId, mPlacementId)) {
-      String adapterError = createAdapterError(ERROR_INVALID_SERVER_PARAMETERS,
-          "Missing or Invalid server parameters.");
-      Log.e(TAG, "Failed to load ad: " + adapterError);
-      if (mMediationInterstitialListener != null) {
-        mMediationInterstitialListener
-            .onAdFailedToLoad(UnityAdapter.this, ERROR_INVALID_SERVER_PARAMETERS);
-      }
+    if (!areValidIds(gameId, mPlacementId)) {
+      sendAdFailedToLoad(ERROR_INVALID_SERVER_PARAMETERS, "Missing or invalid server parameters.");
       return;
     }
 
     if (context == null || !(context instanceof Activity)) {
-      String adapterError =
-          createAdapterError(
-              ERROR_CONTEXT_NOT_ACTIVITY, "Unity Ads requires an Activity context to load ads.");
-      Log.e(TAG, "Failed to load ad: " + adapterError);
-      if (mMediationInterstitialListener != null) {
-        mMediationInterstitialListener
-            .onAdFailedToLoad(UnityAdapter.this, ERROR_CONTEXT_NOT_ACTIVITY);
-      }
+      sendAdFailedToLoad(ERROR_CONTEXT_NOT_ACTIVITY, "Unity Ads requires an Activity context to load ads.");
```

Comment:
I wonder about those checks, dont we have similar checks on the SDK side?

Reply:
we do but it looks like this is done in every adapter by other companies, can't hurt to prevent extra work when we know it will fail? and allows us to directly report the specific error to AdMob